lib/gollum-lib/sanitization.rb

```diff
@@ -1,378 +1,52 @@
+    def self.accepted_protocols
+      @@accepted_protocols
+    end
 
-    # Default whitelisted elements.
-    ELEMENTS   = ([
-      'a', 'abbr', 'acronym', 'address', 'area', 'b', 'big',
-      'blockquote', 'br', 'button', 'caption', 'center', 'cite',
-      'code', 'col', 'colgroup', 'dd', 'del', 'dfn', 'dir', 'div',
-      'dl', 'dt', 'em', 'fieldset', 'font', 'form', 'h1', 'h2', 'h3',
-      'h4', 'h5', 'h6', 'hr', 'i', 'img', 'input', 'ins', 'kbd', 'label',
-      'legend', 'li', 'map', 'mark', 'math', 'menu', 'mfrac', 'mi', 'mn',
-      'mo', 'mrow', 'msqrt', 'msubsup', 'msup', 'mtext', 'ol', 'optgroup',
-      'option', 'p', 'pre', 'q', 's', 'samp', 'select', 'small', 'span',
-      'strike', 'strong', 'sub', 'sup', 'table', 'tbody', 'td', 'textarea',
-      'tfoot', 'th', 'thead', 'tr', 'tt', 'u', 'ul', 'var'
-    ] + MATHML_ELEMENTS).freeze
+    REMOVE_NODES = ['style', 'script']
 
+    SCRUB_REMOVE = Loofah::Scrubber.new do |node|
+      node.remove if REMOVE_NODES.include?(node.name)
+    end
 
-    # Default whitelisted attributes.
-    ATTRIBUTES = ({
-        'a'   => ['href'],
-        'img' => ['src'],
-        :all  => ['abbr', 'accept', 'accept-charset',
-                  'accesskey', 'action', 'align', 'alt', 'axis',
-                  'border', 'cellpadding', 'cellspacing', 'char',
-                  'charoff', 'class', 'charset', 'checked', 'cite',
-                  'clear', 'cols', 'colspan', 'color',
-                  'compact', 'coords', 'datetime', 'dir',
-                  'disabled', 'enctype', 'for', 'frame',
-                  'headers', 'height', 'hreflang',
-                  'hspace', 'id', 'ismap', 'label', 'lang',
-                  'longdesc', 'maxlength', 'media', 'method',
-                  'multiple', 'name', 'nohref', 'noshade',
-                  'nowrap', 'prompt', 'readonly', 'rel', 'rev',
-                  'rows', 'rowspan', 'rules', 'scope',
-                  'selected', 'shape', 'size', 'span',
-                  'start', 'summary', 'tabindex', 'target',
-                  'title', 'type', 'usemap', 'valign', 'value',
-                  'vspace', 'width']
-    }.merge(MATHML_ATTRS)).freeze
+    attr_reader :id_prefix
 
-    # Default whitelisted protocols for URLs.
-    PROTOCOLS  = {
-        'a'    => { 'href' => ['http', 'https', 'mailto', 'ftp', 'irc', 'apt', :relative] },
-        'img'  => { 'src' => ['http', 'https', :relative] },
-        'form' => { 'action' => ['http', 'https', :relative] }
-    }.freeze
+    def initialize(to_xml_opts = {})
+      @to_xml_opts = to_xml_opts
+    end
 
-    ADD_ATTRIBUTES  = lambda do |env, node|
-      if (add = env[:config][:add_attributes][node.name])
-        add.each do |key, value|
-          node[key] = value
-        end
-      end
+    def clean(data, historical = false)
+      doc = Loofah.fragment(data)
+      doc.scrub!(SCRUB_REMOVE)
+      doc.scrub!(:strip)
+      doc.scrub!(:nofollow) if historical
```

Comment:
Loofah's built-in `:nofollow` scrubber allows a nice simplification of the 'history sanitization' logic.

Reply:
👍